great_expectations_cloud/agent/actions/generate_schema_change_expectations_action.py

```diff
@@ -52,30 +58,87 @@ def __init__(  # noqa: PLR0913  # Refactor opportunity
+                # this logic will change with ZELDA-1154 and we will be pulling this in from a mapping
+                expectation_suite_name = f"{data_asset.name} - Expectation Suite"
+                expectation = self._add_schema_change_expectation(
+                    metric_run, expectation_suite_name
+                )
+                created_resources.append(
+                    CreatedResource(resource_id=str(metric_run_id), type="MetricRun")
+                )
+                created_resources.append(
+                    CreatedResource(resource_id=expectation.id, type="Expectation")
+                )
+            except Exception as e:
+                # TODO - follow up in ZELDA-1153
+                LOGGER.warning(f"asset_name: {asset_name} failed with error: {e}")
+                pass
         return ActionResult(
             id=id,
             type=event.type,
-            created_resources=[
-                CreatedResource(resource_id=str(metric_run_id), type="MetricRun"),
-            ],
+            created_resources=created_resources,
         )
 
+    def _retrieve_asset_from_asset_name(
+        self, event: GenerateSchemaChangeExpectationsEvent, asset_name: str
+    ) -> DataAsset:
+        try:
+            datasource = self._context.data_sources.get(event.datasource_name)
+            data_asset = datasource.get_asset(asset_name)
+            data_asset.test_connection()  # raises `TestConnectionError` on failure
+
+        except Exception as e:
+            # TODO - see if this can be made more specific
+            raise RuntimeError(f"Failed to retrieve asset: {e}") from e  # noqa: TRY003 # want to keep this informative for now
+
+        return data_asset
+
+    def _get_metrics(self, data_asset: DataAsset) -> tuple[MetricRun, UUID]:
+        batch_request = data_asset.build_batch_request()
+        metric_run = self._batch_inspector.compute_metric_list_run(
+            data_asset_id=data_asset.id,
+            batch_request=batch_request,
+            metric_list=[MetricTypes.TABLE_COLUMNS, MetricTypes.TABLE_COLUMN_TYPES],
+        )
+        metric_run_id = self._metric_repository.add_metric_run(metric_run)
+        # Note: This exception is raised after the metric run is added to the repository so that
+        # the user can still access any computed metrics even if one of the metrics fails.
+        self._raise_on_any_metric_exception(metric_run)
+
+        return metric_run, metric_run_id
+
+    def _add_schema_change_expectation(
+        self, metric_run: MetricRun, expectation_suite_name: str
+    ) -> gx_expectations.Expectation:
+        try:
+            expectation_suite = self._context.suites.get(name=expectation_suite_name)
+        except DataContextError as e:
+            raise RuntimeError(  # noqa: TRY003 # want to keep this informative for now
+                f"Expectation Suite with name {expectation_suite_name} was not found."
```

Comment:
In the event we don't find the suite, is there anything in place that creates one?

Reply:
Hey good question. I took that out because of the SUJE work, which will create a default GX-managed ExpectationSuite along with the asset.  

The format will look like this. 
```
<data_asset_name> <random 6 char hash> - GX-Managed Checkpoint"
"<data_asset_name> <random 6 char hash> - GX-Managed Expectation Suite"
```

So the appropriate response here would be to report the error back (as part of the ZELDA-1153 and ZELDA-1127)

Reply:
Awesome, thanks for clarifying!